p2p/ipld/plugin/nodes/nodes.go

```diff
@@ -223,10 +225,12 @@ func (n *NmtNodeAdder) Visit(hash []byte, children ...[]byte) {
 	cid := mustCidFromNamespacedSha256(hash)
 	switch len(children) {
 	case 1:
-		n.batch.Add(n.ctx, nmtLeafNode{
-			cid:  cid,
-			Data: children[0],
-		})
+		if n.leaves.Visit(cid) {
```

Comment:
This is cool! Wouldn't it make sense to check the same for leaves as well? If you look at the extended data square, e.g. the leaf at (0, 0) is contained in the first row and the first column. So with your changes, it seems it should make sense as there is only one batchAdder per extended block now?

Reply:
> wouldn't it make sense to check the same for leaves as well (case 2 below)?

Yeah, it makes sense now, but only for leaf nodes, not non-leaf ones(case 2).

Reply:
From my understanding we should not get any duplicates for case 2

Reply:
Friendly ping @liamsi to understand if this is resolved

Reply:
You are absolutely right!